src/main/webapp/dev/js/common/instructorEnroll.js

```diff
@@ -0,0 +1,160 @@
+ *  TestSection2||TestName2|test2@example.com|\n"
+ *
+ * @param spreadsheetData
+ * @returns {string} user data rows
+ */
+function getUserDataRows(spreadsheetData) {
+    // needs to check for '' as an initial empty row with null values will be converted to e.g. "||||" after .map
+    return spreadsheetData.filter(row => (!row.every(cell => cell === null || cell === '')))
+            .map(row => row.join('|'))
+            .join('\n');
+}
+
+/**
+ * Pushes data from spreadsheetDataRows into an array.
+ * Facilitates the function loadData for the Handsontable instance.
+ *
+ * @param spreadsheetDataRows
+ * @returns {Array} updated data
+ */
+function getUpdatedData(spreadsheetDataRows) {
+    return spreadsheetDataRows.map(row => row.split('|'));
+}
+
+/**
+ * Transforms the first uppercase letter of a string into a lowercase letter.
+ * @param string
+ * @returns {string} Handsontable column header in all lowercase letters
+ */
+function unCapitalizeFirstLetter(string) {
+    return string.charAt(0).toLowerCase() + string.slice(1);
+}
+
+/**
+ * Converts returned AJAX data to a suitable format required by Handsontable.
+ * @param studentsData
+ * @param handsontableColHeader
+ * @returns {Array} student data
+ */
+function ajaxDataToHandsontableData(studentsData, handsontableColHeader) {
+    const headers = handsontableColHeader.map(unCapitalizeFirstLetter);
+    return studentsData.map(student => (headers.map(
+            header => student[header])));
+}
+
+/**
+ * Toggle the chevron image depending on the user's action.
+ * @param panelCollapse
+ * @param toggleChevron
+ */
+function toggleChevronImage(panelCollapse, toggleChevron) {
+    if ($(panelCollapse).attr('class').indexOf('checked') === -1) {
+        $(toggleChevron).addClass('glyphicon-chevron-down').removeClass('glyphicon-chevron-up');
+    } else {
+        $(toggleChevron).addClass('glyphicon-chevron-up').removeClass('glyphicon-chevron-down');
+    }
+}
+
+/**
+ * Shows the "Existing students" panel with the spreadsheet interface.
+ * @param $panelHeading
+ * @param panelCollapse
+ */
+function expandStudentsPanel($panelHeading, panelCollapse) {
+    $(panelCollapse).collapse('show');
+    $(panelCollapse[0]).addClass('checked');
+}
+
+/**
+ * Hides the "Existing students" panel.
+ * @param $panelHeading
+ * @param panelCollapse
+ */
+function collapseStudentsPanel($panelHeading, panelCollapse) {
+    $(panelCollapse[0]).collapse('hide');
+    $(panelCollapse[0]).removeClass('checked');
+}
+
+/**
+ * Expands/Collapses the "Existing students" panel depending on the current state of the panel.
+ * @param $panelHeading
+ * @param panelCollapse
+ * @param displayIcon
+ * @param toggleChevron
+ */
+/*  eslint no-unused-expressions: [2, { allowTernary: true }]   */
+function toggleStudentsPanel($panelHeading, panelCollapse, displayIcon, toggleChevron) {
+    displayIcon.html('');
+    ($(panelCollapse[0]).attr('class').indexOf('checked') === -1) ?
```

Comment:
Rather than doing this and creating an unnecessary ESLint suppression, why not just:
```js
if ($(panelCollapse[0]).attr('class').indexOf('checked') === -1)) {
    expandStudentsPanel($panelHeading, panelCollapse)
} else {
    collapseStudentsPanel($panelHeading, panelCollapse)
}
```

Reply:
Yes. The linter highlight this for a reason. The ternary expression is not meant to be used in this way.